packages/dds/merge-tree/src/sortedSegmentSet.ts

```diff
@@ -4,19 +4,12 @@
  */
 
 import { ISegment } from "./mergeTreeNodes";
+import { combineOverlapClients, PartialSequenceLength } from "./partialLengths";
 
-/**
- * Stores a unique and sorted set of segments, or objects with segments
- *
- * This differs from a normal sorted set in that the keys are not fixed.
- * The segments are sorted via their ordinals which can change as the merge tree is modified.
- * Even though the values of the ordinals can change their ordering and uniqueness cannot, so the order of a set of
- * segments ordered by their ordinals will always have the same order even if the ordinal values on
- * the segments changes. This invariant allows ensure the segments stay ordered and unique, and that new segments
- * can be inserted into that order.
- */
-export class SortedSegmentSet<T extends ISegment | { readonly segment: ISegment; } = ISegment> {
-    private readonly ordinalSortedItems: T[] = [];
+export abstract class SortedSet<T, U extends string | number> {
```

Comment:
i think it would be best to put this in a new file, and only leave sorted segment set here